test/no-useless-undefined.js

```diff
@@ -0,0 +1,202 @@
+import test from 'ava';
+import {outdent} from 'outdent';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/no-useless-undefined';
+
+const messageId = 'no-useless-undefined';
+
+const ruleTester = avaRuleTester(test, {
+	parserOptions: {
+		ecmaVersion: 2020
+	}
+});
+
+const typescriptRuleTester = avaRuleTester(test, {
+	parser: require.resolve('@typescript-eslint/parser')
+});
+
+const errors = [{messageId}];
+
+ruleTester.run('no-useless-undefined', rule, {
+	valid: [
+		'function foo() {return;}',
+		'const foo = () => {};',
+		'let foo;',
+		'var foo;',
+		'const foo = undefined;',
+		'foo();',
+		'foo(bar,);',
+		'foo(undefined, bar);',
+		'const {foo} = {};',
+		'function foo({bar} = {}) {}',
+		'function foo(bar) {}'
+	],
+	invalid: [
+		{
+			code: 'function foo() {return undefined;}',
+			output: 'function foo() {return ;}',
```

Comment:
I assume the space between `return` and `;` is intentional? And that the intention is that other style rules would fix it.

Reply:
I simply remove the `undefined` node for `return` and `yield`  cases, so the space is not taking care, do you want me to remove it?

Reply:
Yes

Reply:
I tested `xo`, [`semi-spacing`](https://eslint.org/docs/rules/semi-spacing) can take care both `return ;` and `yield ;`, but I'm going to remove the space here.

Reply:
Done in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/718/commits/6f4b355a1c276d7bc7007168c9653d2329ece00d